From 6fab971870440dfbbd0ad786ec744ccd1b33aa80 Mon Sep 17 00:00:00 2001
From: "johan.bohlin" <johan.bohlin@intel.com>
Date: Mon, 29 Sep 2014 16:53:37 +0200
Subject: [PATCH 1/9] ASoC: Intel: Add IN3 Boost and Differential Mode
 controls

Added alsa controls for setting IN3 Boost and Single
Ended/Differential mode. Initialized IN3 Boost to 1
and IN3 as differential mode.

Issue: GMINL-708, GMINL-1138, GMINL-1139
Change-Id: I8149f9375117819baa99487fc09a1ca7f20916d9
Signed-off-by: johan.bohlin <johan.bohlin@intel.com>
---
 sound/soc/codecs/rt5640.c |   11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/rt5640.c b/sound/soc/codecs/rt5640.c
index 8dc51d8..d8a2b34 100644
--- a/sound/soc/codecs/rt5640.c
+++ b/sound/soc/codecs/rt5640.c
@@ -101,8 +101,9 @@ static struct rt5640_init_reg init_list[] = {
 /*	{RT5640_SPO_R_MIXER	, 0x1800},//DAC -> SPORMIX*/
 /*	{RT5640_I2S1_SDP	, 0xD000},//change IIS1 and IIS2*/
 	/*record */
-	{RT5640_IN1_IN2, 0x5080},	/*IN1 boost 40db and differential mode */
-	{RT5640_IN3_IN4, 0x0000},	/*IN2 boost 40db and signal ended mode */
+	{RT5640_IN1_IN2, 0x51C0},	/*IN1 boost 40db, IN3 20db
+					and differential mode */
+	{RT5640_IN3_IN4, 0x0000},	/*IN2 boost 0db and single ended mode */
 /*	{RT5640_REC_L2_MIXER	, 0x007d},//Mic1 -> RECMIXL*/
 /*	{RT5640_REC_R2_MIXER	, 0x007d},//Mic1 -> RECMIXR*/
 	{RT5640_REC_L2_MIXER, 0x006f},	/*Mic2 -> RECMIXL */
@@ -753,6 +754,9 @@ static const SOC_ENUM_SINGLE_DECL(rt5640_in1_mode_enum, RT5640_IN1_IN2,
 static const SOC_ENUM_SINGLE_DECL(rt5640_in2_mode_enum, RT5640_IN3_IN4,
 				  RT5640_IN_SFT2, rt5640_input_mode);
 
+static const SOC_ENUM_SINGLE_DECL(rt5640_in3_mode_enum, RT5640_IN1_IN2,
+				  RT5640_IN_SFT2, rt5640_input_mode);
+
 /* Interface data select */
 static const char * const rt5640_data_select[] = {
 	"Normal", "Swap", "left copy to right", "right copy to left"
@@ -897,6 +901,9 @@ static const struct snd_kcontrol_new rt5640_snd_controls[] = {
 	SOC_ENUM("IN2 Mode Control", rt5640_in2_mode_enum),
 	SOC_SINGLE_TLV("IN2 Boost", RT5640_IN3_IN4,
 		       RT5640_BST_SFT2, 8, 0, bst_tlv),
+	SOC_ENUM("IN3 Mode Control", rt5640_in3_mode_enum),
+	SOC_SINGLE_TLV("IN3 Boost", RT5640_IN1_IN2,
+		       RT5640_BST_SFT2, 8, 0, bst_tlv),
 	SOC_SINGLE_TLV("Speaker Boost", RT5640_SPO_CLSD_RATIO,
 		       0, 7, 0, speaker_boost_tlv),
 	/* INL/INR Volume Control */
-- 
1.7.9.5

